recipes/dataframe/all/conanfile.py

```diff
@@ -0,0 +1,86 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.tools import Version
+from conans.errors import ConanInvalidConfiguration
+
+
+class DataFrameConan(ConanFile):
+    name = "dataframe"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/hosseinmoein/DataFrame"
+    description = "C++ DataFrame -- R's and Pandas DataFrame in modern C++ using native types, continuous memory storage, and no virtual functions"
+    topics = (
+        "conan",
+        "dataframe",
+        "numerical-analysis",
+        "multidimensional-data",
+        "heterogeneous",
+    )
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+
+    @property
+    def _source_subfolder(self):
+        return os.path.join(self.source_folder, "source_subfolder")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = "DataFrame-V-" + self.version
+        os.rename(extracted_folder, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        compiler = self.settings.compiler
+        if (
+            compiler == "Visual Studio"
+            and Version(self.settings.compiler.version) < "16"
+        ):
+            raise ConanInvalidConfiguration("DataFrame requires Visual Studio >= 16")
+        elif compiler == "apple-clang":
+            raise ConanInvalidConfiguration("DataFrame is not tested with apple-clang")
```

Comment:
What do you mean by "not tested"? Who didn't test, you or the author? You should check if it's working on OSX.

Reply:
OSX tests are commented out in the official [hosseinmoein/DataFrame/.travis.yml](https://github.com/hosseinmoein/DataFrame/blob/master/.travis.yml).
emm, I will test this in my fork with travis-ci.